pkg/volume/util/util_windows.go

```diff
@@ -0,0 +1,30 @@
+// +build windows
+
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"os"
+
+	"github.com/hectane/go-acl"
+	//"golang.org/x/sys/windows"
+)
+
+func Chmod(name string, mode os.FileMode) error {
+	return acl.Chmod(name, mode)
```

Comment:
thanks for the PR @claudiubelu 

i had a look at hectane/go-acl and here is a short summary.

- it looks well written
- license = MIT
- it's only dependency is golang.org/x/sys/windows
- has no release tags, which is mostly problematic for k/k (but not a precedent)
- there are open issues / PRs from 2018 with no response from the maintainers. this is a problematic sign, but at least these seem unrelated to the Chmod usage.
https://github.com/hectane/go-acl/blob/master/go.mod
- it requires advapi32.dll to be present on the host for API calls (not new, similarly to Go's stdlib).
- the mapping between the UNIX permissions and Windows is interesting and i haven't seen this before:
https://github.com/hectane/go-acl/blob/master/chmod.go#L14
our SIG Windows experts should have a good look to see if they agree with them.

one question: depending on who / what runs this code on a Windows host and outside of tests, can Administrators be the only group that can change the permissions? i'm assuming this volume code is used in the kubelet and trying to understand if we are somehow changing the requirements of how the kubelet is run on Windows.

/sig windows storage

Reply:
@marosset also had some concerns with introducing an external library when I was working on https://github.com/kubernetes/kubernetes/pull/104660. So might be good for him to chime in here also.

Reply:
@aravindhp the PR / issue you've linked is actually this PR. What is the actually link you meant?

Reply:
Sorry about that. I have fixed the link.

Reply:
@aravindhp are you sure it was me who had an issue introducing a new library?

As @neolit123 pointed out the way they map UNIX-like permissions to Windows permissions is interesting and I want to take a deeper look at the library.

I am used to using the built in tool [icacls](https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/icacls) to modify permissions. It is not the easiest tool to use but it is very powerful.

> can Administrators be the only group that can change the permissions?

On Windows file "owners" can set permissions on files they own even if they aren't administrators. Administrators can assign ownership of files to non-admin users with icacls /setowner. 
Admins can almost always change file permissions too (unless the files are owned by TrustedInstalled which is reserved for modifying/updating Windows system components).

Reply:
@marosset, sorry my bad. It was @jsturtevant. [Slack reference](https://kubernetes.slack.com/archives/C0SJ4AFB7/p1623794652186700?thread_ts=1623439088.166000&cid=C0SJ4AFB7)

Reply:
> On Windows file "owners" can set permissions on files they own even if they aren't administrators. Administrators can assign ownership of files to non-admin users with icacls /setowner.
Admins can almost always change file permissions too (unless the files are owned by TrustedInstalled which is reserved for modifying/updating Windows system components).

ah, yes i know that non-admin user can also change permissions; didn't know about TrustedInstalled though.

i guess my question was more about whether we are going to somehow surprise our users with an unexpected change to the current set of Windows k8s distributions, such as the one that resides in https://github.com/kubernetes-sigs/sig-windows-tools.

possibly/hopefully not.